tooling/index.js

```diff
@@ -209,6 +209,49 @@ class Oas {
       });
   }
 
+  /**
+   * With a fully composed server URL, run through our list of known OAS servers and return back which server URL was
+   * selected along with any contained server variables split out.
+   *
+   * For example, if you have an OAS server URL of `https://{name}.example.com:{port}/{basePath}`, and pass in
+   * `https://buster.example.com:3000/pet` to this function, you'll get back the following:
+   *
+   *    { selected: 0, variables: { name: 'buster', port: 3000, basePath: 'pet' } }
+   *
+   * Re-supplying this data to `oas.url()` should return the same URL you passed into this method.
+   *
+   * @param {String} baseUrl
+   * @returns {Object|Boolean}
+   */
+  splitVariables(baseUrl) {
```

Comment:
I named this to match `splitUrl` because it's doing similar work `serverVariables` just deals with cases where we don't yet know what the variables in the server URL are. I'm open to suggestions if you have a better idea for a different name.